src/main/webapp/app/course/competencies/competency-accordion/competency-accordion.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { faFilePdf, faList } from '@fortawesome/free-solid-svg-icons';
+import { CompetencyProgress, getConfidence, getIcon, getMastery, getProgress } from 'app/entities/competency.model';
+import { Course } from 'app/entities/course.model';
+import { Router } from '@angular/router';
+import { ICompetencyAccordionToggleEvent } from 'app/shared/competency/interfaces/competency-accordion-toggle-event.interface';
+import { CompetencyInformation, LectureUnitInformation, StudentMetrics } from 'app/entities/student-metrics.model';
+import { round } from 'app/shared/util/utils';
+import { Exercise } from 'app/entities/exercise.model';
+import dayjs from 'dayjs/esm';
+import { lectureUnitIcons, lectureUnitTooltips } from 'app/entities/lecture-unit/lectureUnit.model';
+
+@Component({
+    selector: 'jhi-competency-accordion',
+    templateUrl: './competency-accordion.component.html',
+    styleUrl: './competency-accordion.component.scss',
+})
+export class CompetencyAccordionComponent implements OnChanges {
+    @Input() course?: Course;
+    @Input() competency: CompetencyInformation;
+    @Input() metrics: StudentMetrics;
```

Comment:
either use the type undefined types like `StudentMetrics | undefined` or require the Input `@Input({required: true})`